core/Symbols.cc

```diff
@@ -1536,13 +1536,28 @@ void Symbol::sanityCheck(const GlobalState &gs) const {
     }
 }
 
-ClassOrModuleRef Symbol::enclosingClass(const GlobalState &gs) const {
-    SymbolRef owner = ref(gs);
-    while (!owner.isClassOrModule()) {
-        ENFORCE(owner.exists(), "non-existing owner in enclosingClass");
-        owner = owner.data(gs)->owner;
+ClassOrModuleRef MethodRef::enclosingClass(const GlobalState &gs) const {
+    // Methods can only be owned by classes or modules.
+    return data(gs)->owner.asClassOrModuleRef();
+}
+
+ClassOrModuleRef SymbolRef::enclosingClass(const GlobalState &gs) const {
+    switch (kind()) {
+        case SymbolRef::Kind::ClassOrModule:
+            return asClassOrModuleRef();
```

Comment:
Why is there no `owner` here anymore?

Reply:
I wonder if it makes sense to call `enclosingClass` on a `ClassOrModule` symbol. out of curiosity, can you try to ENFORCE that we never call it on one?

Reply:
Look at the old loop carefully:

```
SymbolRef owner = ref(gs);
while (!owner.isClassOrModule()) {
  //
}
return owner;
```

_It's not returning the owner.__ It's returning itself if it's already a class.

>I wonder if it makes sense to call enclosingClass on a ClassOrModule symbol.

It does! We do this all of the time. The common case is calling `enclosingClass` on `ctx.owner`, which is either a method or a class.

Reply:
Oh, wait, I see your question. Good question.